src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -115,6 +115,21 @@
       (http/error-response (tru "must accept {0}" content-type)
                            http/status-not-acceptable))))
 
+(defn verify-content-encoding
+  "Verification for the specified list of content-encodings."
+  [app content-encodings]
+  {:pre [(coll? content-encodings)
+         (every? string? content-encodings)]}
+  (fn [{:keys [headers request-method] :as req}]
+    (let [content-encoding (headers "content-encoding")]
+      (if (or (not= request-method :post)
+              (empty? content-encoding)
+              (some #{content-encoding} content-encodings))
```

Comment:
oh ha, didn't see that. That's a good enough reason to leave it this way, so I'll leave it up to you.